File: fixtures/scale/dart/large/lib/store/session_log.dart
import 'dart:io';

class SessionLog {
  final String logFile;

  SessionLog(String directory) : logFile = '$directory/session.log';

  void note(String message) {
    try {
      File(logFile).writeAsStringSync('${DateTime.now()} $message\n', mode: FileMode.append);
    } catch (_) {
      // A missing session log never fails the run.
    }
  }
}

File: fixtures/scale/scala/large/src/cryptobench/store/SessionLog.scala
package cryptobench.store

import java.nio.file.Files
import java.nio.file.Path
import java.nio.file.StandardOpenOption
import java.time.Instant

import scala.util.Try

final class SessionLog(directory: Path) {
  private val logFile = directory.resolve("session.log")

  def note(message: String): Unit = {
    val line = Instant.now().toString + " " + message + System.lineSeparator()
    Try(Files.writeString(logFile, line, StandardOpenOption.CREATE, StandardOpenOption.APPEND))
    ()
  }
}
